packages/blockchain-wallet-v4-frontend/src/modals/ViewPrivateKeyWalletFlyout/ViewPrivateKeyWalletFlyout.tsx

```diff
@@ -0,0 +1,315 @@
+        <FormattedMessage
+          id='viewPrivateKeyWalletFlyout.receiveButton.description'
+          defaultMessage='Receive {coin} to your account'
+          values={{ coin }}
+        />
+      }
+    />
+  )
+
+  const swapListTile = (
+    <FlyoutListTile
+      disabled={!hasFunds}
+      onClick={() => {
+        if (!privateKeyWallet || privateKeyWallet.balance === undefined) return
+
+        openSwapModal({
+          balance: privateKeyWallet.balance,
+          baseCoin: privateKeyWallet.coin,
+          coin: privateKeyWallet.coin,
+          label: privateKeyWallet.label,
+          type: SwapBaseCounterTypes.ACCOUNT
+        })
+      }}
+      icon={<Icon name='arrows-horizontal' size='10px' />}
+      iconColor={coinColor}
+      title={
+        <FormattedMessage id='viewPrivateKeyWalletFlyout.swapButton.title' defaultMessage='Swap' />
+      }
+      subtitle={
+        <FormattedMessage
+          id='viewPrivateKeyWalletFlyout.swapButton.description'
+          defaultMessage='Exchange {coin} for Another Crypto'
+          values={{ coin }}
+        />
+      }
+    />
+  )
+
+  return (
+    <Flyout
+      position={position}
+      isOpen={isOpen}
+      userClickedOutside={userClickedOutside}
+      onClose={handleClose}
+      data-e2e='privateKeyModal'
+      total={total}
+    >
+      <div style={{ fontFeatureSettings: "'zero' on, 'ss01' on" }}>
+        <Padding horizontal={40} top={40} bottom={32}>
+          <Flex justifyContent='space-between' alignItems='center' gap={16}>
+            <Flex gap={16} alignItems='center'>
+              <IconCircularBackground color={coinColor} size='large'>
+                <Icon name='key' size='10px' color='white' />
+              </IconCircularBackground>
+
+              <Text weight={600} size='20px' lineHeight='30px' color='grey900'>
+                {privateKeyWallet?.label}
+              </Text>
+            </Flex>
+
+            {closeButton}
+          </Flex>
+        </Padding>
+
+        <Padding horizontal={40} bottom={40}>
+          <Flex flexDirection='column' gap={8}>
+            <Text weight={600} size='24px' lineHeight='32.4px' color='grey900'>
+              {totalFiat}
+            </Text>
+
+            <Text weight={600} size='20px' lineHeight='30px' color='grey700'>
+              {totalCrypto}
+            </Text>
+          </Flex>
+        </Padding>
+
+        {hasFunds ? (
+          <>
+            <Divider />
+            {sendListTile}
+            <Divider />
+            {receiveListTile}
+            <Divider />
+            {swapListTile}
+            <Divider />
+            {sellListTile}
+            <Divider />
+          </>
+        ) : (
+          <>
+            <Divider />
+            {receiveListTile}
+            <Divider />
+            {sendListTile}
+            <Divider />
+            {swapListTile}
+            <Divider />
+            {sellListTile}
+            <Divider />
+          </>
+        )}
```

Comment:
those 2 looks almost identical you can use check of hasFunds on line 286 and prevent duplications

Reply:
I see, I think the problem is that when the user have no funds, we have to show the list with a different order.
So when the user has funds we have to show the send button first and when they have no funds we have to show the receive button first.

Could you show an example of how the code could look like?

Reply:
sure like this


```
        <>
            <Divider />
            {hasFunds ? sendListTile : receiveListTile}
            <Divider />
            {hasFunds ?  receiveListTile: sendListTile}
            <Divider />
            {swapListTile}
            <Divider />
            {sellListTile}
            <Divider />
          </>
```